stdlib/public/core/Dictionary.swift

```diff
@@ -2961,6 +2975,7 @@ final internal class _SwiftDeferredNSDictionary<Key: Hashable, Value>
         // keys null, objects nonnull
         for position in 0..<bucketCount {
           if bridgedBuffer.isInitializedEntry(at: position) {
+            guard i < count else { break }
```

Comment:
you could eke out a one less loop iteration by checking this after you increment i -- and adding an early out above for count == 0

Reply:
Good point! It can often be more than one iteration, too, since the loop skips over empty buckets.